x-pack/plugins/ingest_manager/server/services/epm/kibana/assets/install.ts

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  SavedObject,
+  SavedObjectsBulkCreateObject,
+  SavedObjectsClientContract,
+} from 'src/core/server';
+import { PACKAGES_SAVED_OBJECT_TYPE } from '../../../../../common';
+import * as Registry from '../../registry';
+import { AssetType, KibanaAssetType, AssetReference } from '../../../../types';
+import { deleteKibanaSavedObjectsAssets } from '../../packages/remove';
+import { getInstallationObject, savedObjectTypes } from '../../packages';
+import { saveInstalledKibanaRefs } from '../../packages/install';
+
+type SavedObjectToBe = Required<SavedObjectsBulkCreateObject> & { type: AssetType };
+export type ArchiveAsset = Pick<
+  SavedObject,
+  'id' | 'attributes' | 'migrationVersion' | 'references'
+> & {
+  type: AssetType;
+};
+
+export async function getKibanaAsset(key: string) {
+  const buffer = Registry.getAsset(key);
+
+  // cache values are buffers. convert to string / JSON
+  return JSON.parse(buffer.toString('utf8'));
+}
+
+export function createSavedObjectKibanaAsset(asset: ArchiveAsset): SavedObjectToBe {
+  // convert that to an object
+  return {
+    type: asset.type,
+    id: asset.id,
```

Comment:
I'd need to look but is the `id` of the `logs-*` `metrics-*` etc, is it `metrics-*` or is it a UUID? I think the code before this was assuming the name of the file was the `id` will this cause issues with any of the dashboards or visualizations?

Reply:
The id of the asset should always match the file name, unless I'm mistaken. I think the packages should validate for it as I did recently fix an nginx asset where the ID was copied over from a different asset and did not match.  I can change it back to using the file name if that feels more secure.  As far the kibana index patterns, those are not created with this function but created here: https://github.com/elastic/kibana/blob/master/x-pack/plugins/ingest_manager/server/services/epm/kibana/index_pattern/install.ts#L135 .